pkg/ignition/config.go

```diff
@@ -67,6 +76,28 @@ func File(config *Config) (string, error) {
 			return "", err
 		}
 	}
+	dnsServers := []string{"[Resolve]"}
+	for _, v := range config.DnsServers {
+		dnsEntry := fmt.Sprintf("%s%s", dnsEqualString, v.String())
+		dnsServers = append(dnsServers, dnsEntry)
+	}
+
+	dnsConf := map[string]interface{}{
+		"storage": map[string]interface{}{
+			"files": []interface{}{map[string]interface{}{
+				"path": dnsConfFile,
+				"mode": fileMode,
+				"contents": map[string]interface{}{
+					"inline": strings.Join(dnsServers, "\n"),
+				},
+			}},
+		},
+	}
+
+	// merge dnsConfiguration with ignition content
+	if err := mergo.Merge(ignitionBase, dnsConf, mergo.WithAppendSlice); err != nil {
+		return "", fmt.Errorf("failed to merge dnsServer configuration with igntition content: %w", err)
+	}
```

Comment:
You are right having a check for `dnsServer` `len` here makes sense. We don't want to allocate anything if no `dnsServers` are set.